### DISCUSSION TITLE
Error encountered when using examples

### DISCUSSION THREAD

**Discussion by Miao-code** [named-data/python-ndn#29] [Q&A]
Hello I'm new to ndn and python-ndn and now I ran into some errors in producer.py and comsumer.py in examples. 

After installed NFD and add no_signature=True in producer.py's put_data function(otherwise there's default identity error), producer can successfully send packages to NFD but consumer can't receive it and has timeout error.

Could you explain how to correctly use these example codes?

**Comment by zjkmxy** [comment]
Hello.

> otherwise there's default identity error

You need to setup [ndnsec](https://named-data.net/doc/ndn-cxx/current/manpages/ndnsec-key-gen.html). You can try to run
```bash
ndnsec-key-gen /localhost/operator
```

> Could you explain how to correctly use these example codes?

This is because by default NFD drops DATA packets without a signature. You can try `digest_sha256=True` to use a SHA256 hash if you don't want to sign it with a key.

**Reply by Pesa** [reply]
> This is because by default NFD drops DATA packets without a signature.

Well, if "without a signature" means that the packet lacks a `SignatureValue` element, then yes, such a packet is malformed according to the NDN spec. You should probably use a [NullSignature](https://redmine.named-data.net/projects/ndn-tlv/wiki/NullSignature) if you don't want to sign but still produce a valid packet.

**Reply by zjkmxy** [reply]
Thank you for the link.
